pkg/ipam/allocator.go

```diff
@@ -242,7 +242,7 @@ func (ipam *IPAM) AllocateNextWithExpiration(family, owner string, timeout time.
 					if ipv6Result != nil {
 						ipam.ReleaseIP(ipv6Result.IP)
 					}
-					return
```

Comment:
the `err` variable is assigned on line 237 and this return will return the value of that variable, there's no need to explicitly return `err` here.

Reply:
`return nil, nil, nil` ? I think the `err`  indicates the real cause of the allocation failure.

Reply:
@ywc689 with the `return` as it is in the current code will not return `nil, nil, nil`, it will return `ipv4Result, ipv6Result, err`. With and without the changes in this PR the behavior is exactly the same. You can easily test it by applying the following diff:
```diff
diff --git a/pkg/ipam/allocator.go b/pkg/ipam/allocator.go
index edba820217..0fa92e48f0 100644
--- a/pkg/ipam/allocator.go
+++ b/pkg/ipam/allocator.go
@@ -234,6 +234,7 @@ func (ipam *IPAM) AllocateNextWithExpiration(family, owner string, timeout time.
        if timeout != time.Duration(0) {
                for _, result := range []*AllocationResult{ipv4Result, ipv6Result} {
                        if result != nil {
+                               result.ExpirationUUID, err = ipam.StartExpirationTimer(result.IP, timeout)
                                result.ExpirationUUID, err = ipam.StartExpirationTimer(result.IP, timeout)
                                if err != nil {
                                        if ipv4Result != nil {
```
and verify that the error is returned